borg/archiver.py

```diff
@@ -1514,6 +1553,102 @@ def build_parser(self, args=None, prog=None):
                                help='path to the repository to be upgraded')
 
+        rewrite_epilog = textwrap.dedent("""
+        Rewrites the contents of existing archives.
+
+        --exclude, --exclude-from and PATH have the exact same semantics
+        as in borg create, this means if a PATH is specified the
+        rewrite includes that path and nothing else: PATH does *not* restrict
+        the rewrite to a path.
+
+        --compression recompresses all chunks. Due to how Borg stores compressed size
+        information this might display incorrect information for archives that were not
+        rewritten at the same time.
+        There is no risk of data loss by this. Use --force to recompress chunks already
+        using the specified compression algorithm.
+
+        --chunker-params will re-chunk all files in the archive, this can be
+        used to have upgraded Borg 0.xx or Attic archives deduplicate with
+        Borg 1.x archives.
+
+        Currently the only file status used for --list is 'I' (file/dir included in
+        rewritten archive).
+
+        borg rewrite is signal safe. Send either SIGINT (Ctrl-C on most terminals) or
+        SIGTERM to request termination.
+
+        Use the *exact same* command line to resume the operation later - changing excludes
+        or paths will lead to inconsistencies (changed excludes will only apply to newly
+        processed files/dirs). Changing compression leads to incorrect size information
+        (which does not cause any data loss, but can be misleading).
+
+        USE WITH CAUTION. Permanent data loss by specifying incorrect patterns is possible.
+
+        Note: The archive under rewrite is only removed after the operation completes. The
+              archive that is built during the rewrite exists at the same time at
+              <ARCHIVE>.rewrite.
+
+        Note: When recompressing or (especially) rechunking space usage can be substantial.
+
+        Note: This changes the archive ID.
+        """)
+        subparser = subparsers.add_parser('rewrite', parents=[common_parser],
+                                          description=self.do_rewrite.__doc__,
+                                          epilog=rewrite_epilog,
+                                          formatter_class=argparse.RawDescriptionHelpFormatter,
+                                          help='create backup')
+        subparser.set_defaults(func=self.do_rewrite)
+        subparser.add_argument('--list', dest='output_list',
+                               action='store_true', default=False,
+                               help='output verbose list of items (files, dirs, ...)')
+        subparser.add_argument('-p', '--progress', dest='progress',
+                               action='store_true', default=False,
+                               help='show progress display while rewriting archives')
+        subparser.add_argument('-f', '--force', dest='force_recompress',
```

Comment:
Yeah, I changed the TODO item to "--always-recompress", stuff like "--force" usually means something can/will go wrong, which isn't the case here.